packages/react-native-sdk/src/components/participants/VideoRenderer.tsx

```diff
@@ -0,0 +1,209 @@
+        subscribedVideoLayoutRef.current = undefined;
+      }
+    }
+  }, [sessionId, viewportVisibilityState, muteVideo, call]);
+
+  useEffect(() => {
+    if (!hasJoinedCall && subscribedVideoLayoutRef.current) {
+      // when call is joined again, we want to use the last subscribed dimension to resubscribe
+      pendingVideoLayoutRef.current = subscribedVideoLayoutRef.current;
+      subscribedVideoLayoutRef.current = undefined;
+    }
+  }, [hasJoinedCall]);
+
+  /**
+   * This effect updates the subscription either
+   * 1. when video tracks are published and was unpublished before
+   * 2. when the view's visibility changes
+   * 3. when call was rejoined
+   */
+  useEffect(() => {
+    // NOTE: We only want to update the subscription if the pendingVideoLayoutRef is set
+    const updateIsNeeded = pendingVideoLayoutRef.current;
+
+    if (!updateIsNeeded || !call || !isPublishingVideoTrack || !hasJoinedCall) {
+      return;
+    }
+
+    // NOTE: When the view is not visible, we want to subscribe to audio only.
+    // We unsubscribe their video by setting the dimension to undefined
+    const dimension = !muteVideo ? pendingVideoLayoutRef.current : undefined;
+
+    call.updateSubscriptionsPartial(videoMode, {
+      [sessionId]: { dimension },
+    });
+
+    if (dimension) {
+      subscribedVideoLayoutRef.current = pendingVideoLayoutRef.current;
+      pendingVideoLayoutRef.current = undefined;
+    }
+  }, [
+    call,
+    isPublishingVideoTrack,
+    videoMode,
+    muteVideo,
+    sessionId,
+    hasJoinedCall,
+  ]);
+
+  useEffect(() => {
+    return () => {
+      subscribedVideoLayoutRef.current = undefined;
+      pendingVideoLayoutRef.current = undefined;
+    };
+  }, [videoMode, sessionId]);
+
+  const onLayout: React.ComponentProps<typeof RTCView>['onLayout'] = (
+    event,
+  ) => {
+    const dimension = {
+      width: Math.trunc(event.nativeEvent.layout.width),
+      height: Math.trunc(event.nativeEvent.layout.height),
+    };
+
+    // NOTE: If the participant hasn't published a video track yet,
+    // or the view is not viewable, we store the dimensions and handle it
+    // when the track is published or the video is enabled.
+    if (!call || !isPublishingVideoTrack || !hasJoinedCall) {
+      pendingVideoLayoutRef.current = dimension;
+      return;
+    }
+
+    // NOTE: We don't want to update the subscription if the dimension hasn't changed
+    if (
+      subscribedVideoLayoutRef.current?.width === dimension.width &&
+      subscribedVideoLayoutRef.current?.height === dimension.height
+    ) {
+      return;
+    }
+    call.updateSubscriptionsPartial(videoMode, {
+      [sessionId]: {
+        dimension,
+      },
+    });
+    subscribedVideoLayoutRef.current = dimension;
+    pendingVideoLayoutRef.current = undefined;
+  };
+
+  if (!canShowVideo) {
+    return (
+      <ParticipantVideoFallback participant={participant} onLayout={onLayout} />
```

Comment:
instead of passing an onLayout Prop to both ParticipantVideoFallback and RTCView why not we have a common container and use onLayout there

```
<View onLayout={onLayout}>
  {canShowVideo?  <RTCView /> :  <ParticipantVideoFallback />}
</View>
```

Reply:
Done ✅